components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java

```diff
@@ -0,0 +1,247 @@
+     *
+     * @param requestedResource details of the resource to be registered
+     * @return Response with the status of the registration
+     */
+
+    @Override
+    public Response registerResource(ResourceDetailsDTO requestedResource) {
+
+        if (requestedResource == null) {
+            log.error("Request body cannot be empty.");
+            return Response.status(Response.Status.BAD_REQUEST).build();
+        }
+        try {
+            Resource registerResource = ResourceUtils.getResourceService()
+                    .registerResource(ResourceUtils.getResource(requestedResource));
+            CreateResourceDTO createResourceDTO = ResourceUtils.createResponse(registerResource);
+            return Response.status(Response.Status.CREATED).entity(createResourceDTO).build();
+        } catch (UMAServiceException e) {
+            handleErrorResponse(e, log);
+        } catch (UMAClientException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when registering resource in resource server.", e);
+            }
+            handleErrorResponse(e, log);
+        } catch (Throwable throwable) {
+            log.error("Internal server error occurred. ", throwable);
+        }
+        return null;
+    }
+
+    /**
+     * Retrieve the resource of the given resourceId
+     *
+     * @param resourceId Resourceid of the resource which need to get retrieved
+     * @return Response with the retrieved resource/ retrieval status
+     */
+
+    @Override
+    public Response getResource(String resourceId) {
+
+        Response response = null;
+
+        try {
+            if (!isResourceId(resourceId)) {
+                try {
+                    Resource resourceRegistration = ResourceUtils.getResourceService().getResourceById
+                            (resourceId);
+                    ReadResourceDTO readResourceDTO = ResourceUtils.readResponse(resourceRegistration);
+                    return Response.status(Response.Status.OK).entity(readResourceDTO).build();
+                } catch (UMAServiceException e) {
+                    handleErrorResponse(e, log);
+                } catch (Throwable throwable) {
+                    log.error("Internal server error occurred. ", throwable);
+                }
+
+            }
+        } catch (UMAClientException e) {
+            handleErrorResponse(e, log);
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when retrieving resource from the resource server.", e);
+            }
+        }
+        return response;
+    }
+
+    /**
+     * Retrieve the available resourceId list
+     *
+     * @param resourceOwnerId
+     * @return Response with the retrieved resourceId's/ retrieval status
+     */
+    @Override
+    public Response getResourceIds(String resourceOwnerId) {
+
+        // For testing purpose current resource owner id is taken to validate pat access token
```

Comment:
When commenting start with caps and end with a full stop. Please follow that pattern for all comments.